pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -0,0 +1,625 @@
+    optional NamespaceOption namespace_options = 2;
+}
+
+message CreatePodSandboxRequest {
+    // The configuration for creating a PodSandBox.
+    optional PodSandboxConfig config = 1;
+}
+
+message CreatePodSandboxResponse {
+    // The id of the PodSandBox
+    optional string pod_sandbox_id = 1;
+}
+
+message DeletePodSandboxRequest {
+    // The id of the PodSandBox
+    optional string pod_sandbox_id = 1;
+}
+
+message DeletePodSandboxResponse {}
+
+message PodSandboxStatusRequest {
+    // The id of the PodSandBox
+    optional string pod_sandbox_id = 1;
+}
+
+// PodSandboxNetworkStatus is the status of the network for a PodSandbox.
+message PodSandboxNetworkStatus {
+    // The IP address of the PodSandbox
+    optional string ip = 1;
+}
+
+// Namespace contains paths to the namespaces.
+message Namespace {
+    // Network is the path to the network namespace.
+    optional string network = 1;
+    // Options is the namespace options for linux namespaces
+    optional NamespaceOption options = 2;
+}
+
+// LinuxSandBoxStatus contains status specific to Linux sandboxes.
+message LinuxPodSandboxStatus {
+    // Namespaces contains paths to the sandbox's namespaces.
+    optional Namespace namespaces = 1;
+}
+
+enum PodSandBoxState {
+    READY = 0;
+    NOTREADY = 1;
+}
+
+// PodSandboxStatus contains the status of the PodSandbox.
+message PodSandboxStatus {
+    // ID of the sandbox.
+    optional string id = 1;
+    // Name of the sandbox
+    optional string name = 2;
+    // State of the sandbox.
+    optional PodSandBoxState state = 3;
+    // Creation timestamp of the sandbox
+    optional int64 created_at = 4;
+    // Network contains network status if network is handled by the runtime.
+    optional PodSandboxNetworkStatus network = 5;
+    // Linux specific status to a pod sandbox.
+    optional LinuxPodSandboxStatus linux = 6;
+    // Labels are key value pairs that may be used to scope and select individual resources.
+    map<string, string> labels = 7;
+    // Annotations is an unstructured key value map that may be set by external
+    // tools to store and retrieve arbitrary metadata.
+    map<string, string> annotations = 8;
+}
+
+message PodSandboxStatusResponse {
+    // The status of the PodSandbox
+    optional PodSandboxStatus status = 1;
+}
+
+// PodSandboxFilter is used to filter a list of PodSandboxes.
+message PodSandboxFilter {
+    // Name of the sandbox.
+    optional string name = 1;
+    // ID of the sandbox.
+    optional string id = 2;
+    // State of the sandbox.
+    optional PodSandBoxState state = 3;
+    // LabelSelector to select matches.
+    // Only api.MatchLabels is supported for now and the requirements
+    // are ANDed. MatchExpressions is not supported yet.
+    map<string, string> label_selector = 4;
+}
+
+message ListPodSandboxRequest {
+    // PodSandboxFilter to filter a list of PodSandboxes.
+    optional PodSandboxFilter filter = 1;
```

Comment:
Or maybe not, as I checked for rkt's filters, we using the label_selector is probably sufficient.